ibm/service/cis/resource_ibm_cis_ruleset.go

```diff
@@ -0,0 +1,651 @@
+						Type:        schema.TypeMap,
+						Optional:    true,
+						Description: "Logging of the Rulesets Rule",
+						Elem:        &schema.Schema{Type: schema.TypeBool},
+					},
+					CISRulesetsRulePosition: {
+						Type:        schema.TypeSet,
+						Optional:    true,
+						Description: "Position of Rulesets Rule",
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								CISRulesetsRulePositionAfter: {
+									Type:        schema.TypeString,
+									Optional:    true,
+									Description: "Ruleset before Position of Rulesets Rule",
+								},
+								CISRulesetsRulePositionBefore: {
+									Type:        schema.TypeString,
+									Optional:    true,
+									Description: "Ruleset after Position of Rulesets Rule",
+								},
+								CISRulesetsRulePositionIndex: {
+									Type:        schema.TypeInt,
+									Optional:    true,
+									Description: "Index of Rulesets Rule",
+								},
+							},
+						},
+					},
+					CISRulesetsRuleLastUpdatedAt: {
+						Type:        schema.TypeString,
+						Optional:    true,
+						Description: "Rulesets Rule Last Updated At",
+					},
+				},
+			},
+		},
+	},
+}
+
+func ResourceIBMCISRuleset() *schema.Resource {
+	return &schema.Resource{
+		Read:     ResourceIBMCISRulesetRead,
+		Update:   ResourceIBMCISRulesetUpdate,
+		Delete:   ResourceIBMCISRulesetDelete,
+		Create:   ResourceIBMCISRulesetRead,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS instance crn",
+				Required:    true,
+				ValidateFunc: validate.InvokeValidator("ibm_cis_ruleset",
+					"cis_id"),
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Optional:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			CISRulesetsId: {
+				Type:        schema.TypeString,
+				Description: "Associated Ruleset ID",
+				Optional:    true,
+			},
+			CISRulesetsObjectOutput: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Container for response information.",
+				Elem:        CISResourceResponseObject,
+			},
+		},
+	}
+}
+func ResourceIBMCISRulesetValidator() *validate.ResourceValidator {
+	validateSchema := make([]validate.ValidateSchema, 0)
+	validateSchema = append(validateSchema,
+		validate.ValidateSchema{
+			Identifier:                 "cis_id",
+			ValidateFunctionIdentifier: validate.ValidateCloudData,
+			Type:                       validate.TypeString,
+			CloudDataType:              "resource_instance",
+			CloudDataRange:             []string{"service:internet-svcs"},
+			Required:                   true})
+	ibmCISRulesetValidator := validate.ResourceValidator{
+		ResourceName: "ibm_cis_ruleset",
+		Schema:       validateSchema}
+	return &ibmCISRulesetValidator
+}
+
+func ResourceIBMCISRulesetCreate(d *schema.ResourceData, meta interface{}) error {
+	return nil
```

Comment:
`d.IsNewResource()` // check if it is a new resource, if true then return error that user need to import it first